chef/automate
# Add modal prompt to delete integration and associated nodes

### :nut_and_bolt: Description: What code changed, and why?
This branch adds two new elements to deleting integrations.
1.  Upon choosing to delete an integration from the integrations list, the user is now prompted to check that they are sure.
2. Nodes associated with the deleted integration will also be deleted.

### :chains: Related Resources
Initial Writeup: https://github.com/chef/automate/issues/4176
Fixes: https://github.com/chef/automate/issues/4335

### :+1: Definition of Done
Upon choosing to delete a node integration, the user is first prompted with a modal.  Once finally deleted, all associated nodes are also deleted.

### :athletic_shoe: How to Build and Test the Change
in root: `build components/automate-ui-devproxy && start_all_services`
in components/automate-ui: `make serve`

To Test:

1. add an aws-ec2 integration (follow these steps https://github.com/chef/automate/blob/master/dev-docs/adding-data/adding_test_data.md#adding-cloud-integrations) - this will add a number of nodes to the system, associating them with that nodemanager so that scan jobs can be executed
2. go to Compliance > Scan Jobs > Nodes Added ==> Note the nodes added to the system as a result.
2. go back to Settings > Integrations,  and delete the nodemanager that you created in step 1.
3. go to back Compliance > Scan Jobs > Nodes Added
4. you will see all the nodes previously created are now deleted as well

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
![DeleteAssociatedNodes](https://user-images.githubusercontent.com/16737484/93383847-10ec1680-f819-11ea-8817-62544bf7e474.gif)
